code/game/machinery/camera/tracking.dm

```diff
@@ -68,57 +68,45 @@
 		trackable_mobs()
 
 	var/datum/weakref/target = (isnull(track.humans[target_name]) ? track.others[target_name] : track.humans[target_name])
+	ai_start_tracking(target.resolve())
 
-	ai_actual_track(target.resolve())
-
-/mob/living/silicon/ai/proc/ai_actual_track(mob/living/target)
-	if(!istype(target))
+/mob/living/silicon/ai/proc/ai_start_tracking(mob/living/target) //starts ai tracking
+	if(!target || !target.can_track(src))
+		to_chat(src, "<span class='warning'>Target is not near any active cameras.</span>")
 		return
-	var/mob/living/silicon/ai/U = usr
-
-	U.cameraFollow = target
-	U.tracking = 1
-
-	if(!target || !target.can_track(usr))
-		to_chat(U, "<span class='warning'>Target is not near any active cameras.</span>")
-		U.cameraFollow = null
+	if(ai_tracking_target) //if there is already a tracking going when this gets called makes sure the old tracking gets stopped before we register the new signals
+		ai_stop_tracking()
+	RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/ai_stop_tracking)
+	RegisterSignal(target, COMSIG_MOVABLE_MOVED, .proc/ai_actual_track)
+	ai_tracking_target = target
+	eyeobj.setLoc(get_turf(target)) //on the first call of this we obviously need to jump to the target ourselfs else we would go there only after they moved once
+	to_chat(src, "<span class='notice'>Now tracking [target.get_visible_name()] on camera.</span>")
+
+/mob/living/silicon/ai/proc/ai_stop_tracking(var/reacquire_failed = FALSE) //stops ai tracking
+	SIGNAL_HANDLER //this can technically be called by a signal too if the target gets qdeleted
+	UnregisterSignal(ai_tracking_target, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(ai_tracking_target, COMSIG_MOVABLE_MOVED)
+	ai_tracking_target = null
+	if(reacquire_timer)
+		if(reacquire_failed) //edge case when someone might jump to another camera while the reacquire timer is running
+			to_chat(src, "<span class='warning'>Unable to reacquire, cancelling track...</span>")
+		else
+			deltimer(reacquire_timer)
```

Comment:
Shouldn't it always delete the timer if you are going to set the timer to null?

Reply:
its not required to delete the timer in the other instance since it will be automatically deleted in that case anyway still need to null the timer id trough